Snowflake.Data.Tests/SFConnectionIT.cs

```diff
@@ -132,13 +134,75 @@ public void TestIncorrectUserOrPasswordBasicConnection()
                 {
                     // Expected
                     logger.Debug("Failed opening connection ", e);
-                    Assert.AreEqual("08006", e.SqlState); // Connection failure
+                    Assert.AreEqual(ConnectionFailureErrorCode, e.SqlState);
                 }
 
                 Assert.AreEqual(ConnectionState.Closed, conn.State);
 			}
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestConnectionIsNotMarkedAsOpenWhenWasNotCorrectlyOpenedBefore(bool explicitClose)
+        {
+            SnowflakeDbConnectionPool.SetPooling(true);
+            SnowflakeDbConnection snowflakeConnection = null;
+            for (var i = 0; i < 2; ++i)

```

Comment:
what do you think of using Enumerable.Range(0, 2) instead of a for loop?

Reply:
I don't see a lot of value in such change - it would add new `using` and replace:
```
for (int i = 0; i < 2; ++i)
```
with 
```
foreach (int i in Enumerable.Range(0, 2))
```
In my opinion is does not increase readability